### ISSUE TITLE
Catch exceptions in Tool

### ISSUE THREAD

**Issue by johnynek** [twitter/scalding#389]
Tool should catch all exceptions and then print suggestions or explanations based on the exception.

For instance: when sources can't find data, we should print at the very bottom:
TL;DR: Data is missing.

Or, if we get a ClassMissing or NoClassDef, or AbstractMethodError, etc... we should print:

GUESS: This may be a problem with the binary version of a dependency. Check which versions of dependencies you're pulling in.

Or: if we get the no-tail connected to head method print something like:

TL;DR: Cascading requires all sources to have final outputs on disk. Source: X is processed but never written.

We can probably find a few more.

Ideally we would have some functions:

def handers: List[(Throwable) => Boolean]

Where we have these functions which can handle some Throwable, and if it does, it returns true, else false.

And then we can do:

``` scala
try {

}
catch {
  case t: Throwable => handlers.find { h => h(t) }
}
```

And it should work.
